pkg/storegateway/gateway_blocks_http.go

```diff
@@ -74,6 +98,33 @@ func (s *StoreGateway) BlocksHandler(w http.ResponseWriter, req *http.Request) {
 	showDeleted := req.Form.Get("show_deleted") == "on"
 	showSources := req.Form.Get("show_sources") == "on"
 	showParents := req.Form.Get("show_parents") == "on"
+	actionType := req.Form.Get("action_type")
+	if actionType == "" {
+		actionType = string(ActionTypeNone) // Set the default value to "none"
+	}
+
+	if !isValidActionType(ActionType(actionType)) {
+		util.WriteTextResponse(w, fmt.Sprintf("Invalid Action Type: %s\n", actionType))
+		return
+	}
+	action := ActionType(actionType)
+
+	blockUlidsString := req.Form.Get("block_ulids")
+	if action != ActionTypeNone && blockUlidsString != "" {
+		var uids []string
+
+		err := json.Unmarshal([]byte(blockUlidsString), &uids)
+		if err != nil {
+			util.WriteTextResponse(w, fmt.Sprintf("Can't decode base64 of selected blocks' uid: %s", err))
+			return
+		}
+		err = s.performActionsOnBlocks(tenantID, req, action, uids)
+		if err != nil {
+			util.WriteTextResponse(w, err.Error())
+			return
+		}
+	}
```

Comment:
If we didn't fail, it would be nice to provide some kind of feedback here.

Reply:
added info text, to clarify what are changed, how many blocks modified